### PR TITLE
docs(compute-samples): added samples and tests for: GCE instance from…

### PR DESCRIPTION
**Pull request by Sita04**
… template

- [ ] I have followed [Sample Format Guide](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md)
- [ ] `pom.xml` parent set to latest `shared-configuration`
- [ ] Appropriate changes to README are included in PR
- [ ] API's need to be enabled to test (tell us)
- [ ] Environment Variables need to be set (ask us to set them)
- [ ] **Tests** pass:   `mvn clean verify` **required**
- [ ] **Lint**  passes: `mvn -P lint checkstyle:check` **required**
- [ ] **Static Analysis**:  `mvn -P lint clean compile pmd:cpd-check spotbugs:check` **advisory only**
- [ ] Please **merge** this PR for me once it is approved.

### CODE REVIEW

**Review by stephaniewang526 — APPROVED** [review]

File: compute/cloud-client/src/main/java/compute/CreateInstanceFromTemplateWithOverrides.java [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+  public static void createInstanceFromTemplateWithOverrides(String projectId, String zone,
+      String instanceName, String instanceTemplateName)
+      throws IOException, ExecutionException, InterruptedException {
+
+    try (InstancesClient instancesClient = InstancesClient.create();
+        ZoneOperationsClient zoneOperationsClient = ZoneOperationsClient.create();
+        InstanceTemplatesClient instanceTemplatesClient = InstanceTemplatesClient.create()) {
+
+      String machineType = "n1-standard-1";
+      String newDiskSourceImage = "projects/debian-cloud/global/images/family/debian-10";
```
Comment (lesv) [inline compute/cloud-client/src/main/java/compute/CreateInstanceFromTemplateWithOverrides.java:73]: debian-11 ??

**Review by lesv — APPROVED** [review]

**Comment by lesv** [comment]
# Java 11
```
#
# compute/cloud-client
#

[ERROR] Tests run: 2, Failures: 0, Errors: 2, Skipped: 0, Time elapsed: 3.382 s <<< FAILURE! - in compute.InstanceTemplatesIT
[ERROR] compute.InstanceTemplatesIT  Time elapsed: 3.355 s  <<< ERROR!
java.util.concurrent.ExecutionException: com.google.api.gax.rpc.PermissionDeniedException: Forbidden
	at gce.diregapic.samples@1.0-SNAPSHOT/compute.InstanceTemplatesIT.setup(InstanceTemplatesIT.java:74)
Caused by: com.google.api.gax.rpc.PermissionDeniedException: Forbidden
Caused by: com.google.api.client.http.HttpResponseException:
403 Forbidden
POST https://compute.googleapis.com:443/compute/v1/projects/java-docs-samples-testing/global/instanceTemplates
{
  "error": {
    "code": 403,
    "message": "Required 'compute.instanceTemplates.create' permission for 'projects/java-docs-samples-testing/global/instanceTemplates/template-name-3c17438d-7948-4088-8ce5-57c794f2bc90'",
    "errors": [
      {
        "message": "Required 'compute.instanceTemplates.create' permission for 'projects/java-docs-samples-testing/global/instanceTemplates/template-name-3c17438d-7948-4088-8ce5-57c794f2bc90'",
        "domain": "global",
        "reason": "forbidden"
      }
    ]
  }
}
```

**Comment by lesv** [comment]
# Java 11
```
#
# compute/cloud-client
#

[ERROR] Tests run: 2, Failures: 0, Errors: 2, Skipped: 0, Time elapsed: 3.382 s <<< FAILURE! - in compute.InstanceTemplatesIT
[ERROR] compute.InstanceTemplatesIT  Time elapsed: 3.355 s  <<< ERROR!
java.util.concurrent.ExecutionException: com.google.api.gax.rpc.PermissionDeniedException: Forbidden
	at gce.diregapic.samples@1.0-SNAPSHOT/compute.InstanceTemplatesIT.setup(InstanceTemplatesIT.java:74)
Caused by: com.google.api.gax.rpc.PermissionDeniedException: Forbidden
Caused by: com.google.api.client.http.HttpResponseException:
403 Forbidden
POST https://compute.googleapis.com:443/compute/v1/projects/java-docs-samples-testing/global/instanceTemplates
{
  "error": {
    "code": 403,
    "message": "Required 'compute.instanceTemplates.create' permission for 'projects/java-docs-samples-testing/global/instanceTemplates/template-name-3c17438d-7948-4088-8ce5-57c794f2bc90'",
    "errors": [
      {
        "message": "Required 'compute.instanceTemplates.create' permission for 'projects/java-docs-samples-testing/global/instanceTemplates/template-name-3c17438d-7948-4088-8ce5-57c794f2bc90'",
        "domain": "global",
        "reason": "forbidden"
      }
    ]
  }
}


[ERROR] compute.InstanceTemplatesIT  Time elapsed: 3.356 s  <<< ERROR!
java.util.concurrent.ExecutionException: com.google.api.gax.rpc.PermissionDeniedException: Forbidden
	at gce.diregapic.samples@1.0-SNAPSHOT/compute.InstanceTemplatesIT.cleanup(InstanceTemplatesIT.java:99)
Caused by: com.google.api.gax.rpc.PermissionDeniedException: Forbidden
Caused by: com.google.api.client.http.HttpResponseException:
403 Forbidden
DELETE https://compute.googleapis.com:443/compute/v1/projects/java-docs-samples-testing/zones/us-central1-a/instances/my-new-test-instance27449ae3-d610-495a-9526-15ca5b51c564
{
  "error": {
    "code": 403,
    "message": "Required 'compute.instances.delete' permission for 'projects/java-docs-samples-testing/zones/us-central1-a/instances/my-new-test-instance27449ae3-d610-495a-9526-15ca5b51c564'",
    "errors": [
      {
        "message": "Required 'compute.instances.delete' permission for 'projects/java-docs-samples-testing/zones/us-central1-a/instances/my-new-test-instance27449ae3-d610-495a-9526-15ca5b51c564'",
        "domain": "global",
        "reason": "forbidden"
      }
    ]
  }
}


[ERROR] Tests run: 2, Failures: 0, Errors: 2, Skipped: 0, Time elapsed: 0.684 s <<< FAILURE! - in compute.SnippetsIT
[ERROR] compute.SnippetsIT  Time elapsed: 0.667 s  <<< ERROR!
java.util.concurrent.ExecutionException: com.google.api.gax.rpc.PermissionDeniedException: Forbidden
	at gce.diregapic.samples@1.0-SNAPSHOT/compute.SnippetsIT.setUp(SnippetsIT.java:209)
Caused by: com.google.api.gax.rpc.PermissionDeniedException: Forbidden
Caused by: com.google.api.client.http.HttpResponseException:
403 Forbidden
POST https://compute.googleapis.com:443/compute/v1/projects/java-docs-samples-testing/zones/us-central1-a/instances
{
  "error": {
    "code": 403,
    "message": "Required 'compute.instances.create' permission for 'projects/java-docs-samples-testing/zones/us-central1-a/instances/my-new-test-instanceb8ccf291-5793-4e06-aa2b-44f4c44c2222'",
    "errors": [
      {
        "message": "Required 'compute.instances.create' permission for 'projects/java-docs-samples-testing/zones/us-central1-a/instances/my-new-test-instanceb8ccf291-5793-4e06-aa2b-44f4c44c2222'",
        "domain": "global",
        "reason": "forbidden"
      },
      {
        "message": "Required 'compute.disks.create' permission for 'projects/java-docs-samples-testing/zones/us-central1-a/disks/my-new-test-instanceb8ccf291-5793-4e06-aa2b-44f4c44c2222'",
        "domain": "global",
        "reason": "forbidden"
      },
      {
        "message": "Required 'compute.subnetworks.use' permission for 'projects/java-docs-samples-testing/regions/us-central1/subnetworks/default'",
        "domain": "global",
        "reason": "forbidden"
      }
    ]
  }
}


[ERROR] compute.SnippetsIT  Time elapsed: 0.673 s  <<< ERROR!
com.google.api.gax.rpc.PermissionDeniedException: Forbidden
Caused by: com.google.api.client.http.HttpResponseException:
403 Forbidden
GET https://compute.googleapis.com:443/compute/v1/projects/java-docs-samples-testing/global/firewalls/firewall-rule-9673629d-30ea-48fe-bc0d-428b07ac7228
{
  "error": {
    "code": 403,
    "message": "Required 'compute.firewalls.get' permission for 'projects/java-docs-samples-testing/global/firewalls/firewall-rule-9673629d-30ea-48fe-bc0d-428b07ac7228'",
    "errors": [
      {
        "message": "Required 'compute.firewalls.get' permission for 'projects/java-docs-samples-testing/global/firewalls/firewall-rule-9673629d-30ea-48fe-bc0d-428b07ac7228'",
        "domain": "global",
        "reason": "forbidden"
      }
    ]
  }
}


[ERROR] Errors:
[ERROR] compute.InstanceTemplatesIT.null
[ERROR]   Run 1: InstanceTemplatesIT.setup:74 » Execution com.google.api.gax.rpc.PermissionDeni...
[ERROR]   Run 2: InstanceTemplatesIT.cleanup:99 » Execution com.google.api.gax.rpc.PermissionDe...
[ERROR] compute.SnippetsIT.null
[ERROR]   Run 1: SnippetsIT.setUp:209 » Execution com.google.api.gax.rpc.PermissionDeniedExcept...
[ERROR]   Run 2: SnippetsIT » PermissionDenied Forbidden
[ERROR] Tests run: 2, Failures: 0, Errors: 2, Skipped: 0
```

File: compute/cloud-client/src/main/java/compute/CreateInstanceFromTemplateWithOverrides.java (hunk 1 above)
Comment (Sita04) [reply compute/cloud-client/src/main/java/compute/CreateInstanceFromTemplateWithOverrides.java]: Done, thanks Les.

**Pull request merged**